src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -1213,7 +1214,11 @@ private string AppendSuffix(string parentId, string suffix, char delimiter)
 
             //generate overflow suffix
             string overflowSuffix = ((int)GetRandomNumber()).ToString("x8");
+#if SPAN_BASED_STRING_CONCAT
+            return string.Concat(parentId.AsSpan(0, trimPosition), overflowSuffix, stackalloc char[] { '#' });
```

Comment:
why you are using stackalloc?

Reply:
There's no implicit conversion from `char` to `ReadOnlySpan<char>`. The alternative is to convert it to a string.

Reply:
> but I guess it makes more sense just to use a string literal.

I agree.